Unable to link against shared libs when using static boost libraries

At least on `amd64` the following error appears after updating to 29e96caf436e37801d7aa9229b746cca64e420a6 in our ci setup
```
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/libboost_system.a(error_code.o): relocation R_X86_64_PC32 against symbol `_ZN5boost6system16generic_categoryEv' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
```

--- a/cmake/Includes/CMakeLists.txt
+++ b/cmake/Includes/CMakeLists.txt
@@ -40,8 +40,13 @@ set(CMAKE_MODULE_PATH
 find_package(Threads REQUIRED)
 find_package(OpenSSL REQUIRED)
 
-# https://stackoverflow.com/a/3176086/884770
-set(Boost_USE_STATIC_LIBS   ON)
+if(BUILD_SHARED_LIBS)
+    set(Boost_USE_STATIC_LIBS OFF )
+else()
+    # https://stackoverflow.com/a/3176086/884770
+    set(Boost_USE_STATIC_LIBS ON )
+endif()
+
 find_package(Boost REQUIRED COMPONENTS program_options system thread random)
 
 find_package(Msgpack REQUIRED)

